fluid/machine_translation.py

```diff
@@ -0,0 +1,329 @@
+    def linear(inputs):
+        return fluid.layers.fc(input=inputs, size=size, bias_attr=True)
+
+    forget_gate = fluid.layers.sigmoid(x=linear([hidden_t_prev, x_t]))
+    input_gate = fluid.layers.sigmoid(x=linear([hidden_t_prev, x_t]))
+    output_gate = fluid.layers.sigmoid(x=linear([hidden_t_prev, x_t]))
+    cell_tilde = fluid.layers.tanh(x=linear([hidden_t_prev, x_t]))
+
+    cell_t = fluid.layers.sums(input=[
+        fluid.layers.elementwise_mul(
+            x=forget_gate, y=cell_t_prev), fluid.layers.elementwise_mul(
+                x=input_gate, y=cell_tilde)
+    ])
+
+    hidden_t = fluid.layers.elementwise_mul(
+        x=output_gate, y=fluid.layers.tanh(x=cell_t))
+
+    return hidden_t, cell_t
+
+
+def seq_to_seq_net(word_vector_dim,
+                   encoder_size,
+                   decoder_size,
+                   source_dict_dim,
+                   target_dict_dim,
+                   is_generating=False,
+                   beam_size=3,
+                   max_length=250):
+    """Construct a seq2seq network."""
+    feeding_list = ["source_sequence", "target_sequence", "label_sequence"]
+
+    def bi_lstm_encoder(input_seq, size):
+        input_forward_proj = fluid.layers.fc(input=input_seq,
+                                             size=size * 4,
+                                             act='tanh')
+        forward, _ = fluid.layers.dynamic_lstm(
+            input=input_forward_proj, size=size * 4)
+        input_reversed_proj = fluid.layers.fc(input=input_seq,
+                                              size=size * 4,
+                                              act='tanh')
+        reversed, _ = fluid.layers.dynamic_lstm(
+            input=input_reversed_proj, size=size * 4, is_reverse=True)
+        return forward, reversed
+
+    src_word_idx = fluid.layers.data(
+        name=feeding_list[0], shape=[1], dtype='int64', lod_level=1)
+
+    src_embedding = fluid.layers.embedding(
+        input=src_word_idx,
+        size=[source_dict_dim, word_vector_dim],
+        dtype='float32')
+
+    src_forward, src_reversed = bi_lstm_encoder(
+        input_seq=src_embedding, size=encoder_size)
+
+    encoded_vector = fluid.layers.concat(
+        input=[src_forward, src_reversed], axis=1)
+
+    encoded_proj = fluid.layers.fc(input=encoded_vector,
+                                   size=decoder_size,
+                                   bias_attr=False)
+
+    backward_first = fluid.layers.sequence_pool(
+        input=src_reversed, pool_type='first')
+
+    decoder_boot = fluid.layers.fc(input=backward_first,
+                                   size=decoder_size,
+                                   bias_attr=False,
+                                   act='tanh')
+
+    def lstm_decoder_with_attention(target_embedding, encoder_vec, encoder_proj,
+                                    decoder_boot, decoder_size):
+        def simple_attention(encoder_vec, encoder_proj, decoder_state):
```

Comment:
The attention mechanism is wrong. Where is 'tanh' operation which appears in original formula?

Reply:
Didn't catch your point, why `tanh` is necessary for attention? There are several kind of attention mechanisms. Please refer to https://github.com/PaddlePaddle/Paddle/blob/9bfa3013891cf3da832307894acff919d6705cee/python/paddle/trainer_config_helpers/networks.py#L1400

Reply:
https://github.com/PaddlePaddle/Paddle/blob/9bfa3013891cf3da832307894acff919d6705cee/python/paddle/trainer_config_helpers/networks.py#L1473   
Here, the mixed_layer performs  `tanh`.  
If you realizes the attention mechanism in` Neural Machine Translation By Jointly Learning To Align and Translate`, tanh is needed.

Reply:
Why do you think it's wrong to apply linear transformation?

Reply:
To keep consistent, will apply `tanh` in next PR. Thanks.

Reply:
Maybe we shouldn't focus on one word. And is the original attention in that paper what you want to realize? Or a modified one?